### DISCUSSION TITLE
try deepsleep using tasmota and wemos d1 mini - problems

### DISCUSSION THREAD

**Discussion by dkotte** [arendst/Tasmota#19660] [Q&A]
Hi all,
currently i try to build a battery monitor using tasmota.
I configured TelePeriod 30 and deepsleeptime 3600
Deep sleep works for some days as expected.
Today for testing I disable deepsleep using deepsleeptime 0 at the console and MQTT. ... and back to 3600.
Afterward, the teleperiod seems ignored.
Device is alive for 2-5 seconds only. It sends no MQTT data.
Back to deepsleeptime 0  , MQTT data is sent correctly.
Someone an idea what's happen .. or an idea how to resolve?
thx
dirk

PS: output from console:
`
00:00:00.002 HDW: ESP8266EX

00:00:00.050 CFG: Loaded from flash at FA, Count 169
00:00:00.076 Project tasmota - TasSleepTest Version 13.1.0(tasmota)-2_7_4_9(2023-08-15T07:58:36)
00:00:00.566 WIF: Connecting to AP1 iD8myIoT Channel 11 BSSId 00:1A:8C:D4:79:40 in mode 11n as TasSleepTest-3348...
00:00:03.760 WIF: Connected
00:00:04.014 HTP: Web server active on TasSleepTest-3348 with IP address 192.168.11.29
18:59:07.015 MQT: Attempting connection...
18:59:07.029 MQT: Connected
18:59:07.032 MQT: fhem/sensors/TasSleepTest/tele/LWT = Online (retained)
18:59:07.034 MQT: fhem/sensors/TasSleepTest/cmnd/POWER = 
18:59:07.039 MQT: fhem/sensors/TasSleepTest/stat/RESULT = {"POWER":"ON"}
18:59:07.041 MQT: fhem/sensors/TasSleepTest/stat/POWER = ON
18:59:07.323 MQT: fhem/sensors/TasSleepTest/stat/RESULT = {"DeepSleepTime":60}
18:59:07.374 MQT: fhem/sensors/TasSleepTest/stat/RESULT = {"POWER":"ON"}
18:59:07.377 MQT: fhem/sensors/TasSleepTest/stat/POWER = ON
18:59:07.449 DSL: Reset wrong settings wakeup: 0, slip 10000
18:59:07.452 MQT: fhem/sensors/TasSleepTest/tele/DEEPSLEEP = {"DeepSleep":{"Time":"2023-10-03T19:00:03","DeepSleep":1696359547,"Wakeup":1696359603}} (retained)`

**Comment by barbudor** [comment]
Why using a Teleperiod of 30 with Deepsleep ?
Have you read and understood the specific behavior of tasmota when using a Teleperiod different from 10 or 300 along with DeepSleep ?

**Reply by dkotte** [reply]
Yes, I've seen this special behavior already.
i try 10 for teleperiod to, but didn't use it, because i need a defined time for trying.
But using teleperiod 10 I didn't get the sensor-data too.

**Comment by dkotte** [comment]
I think I've found the problem.
Since my MQTT doesn't support QOS, I used the MQTT-"retain" flag.
Now after every reboot the ESP has received the “DeepSleepTime” from MQTT... and starts sleeping immediately.

**Reply by sfromis** [reply]
You can break that by removing the retained message.
